src/content/reference/react-dom/components/form.md

```diff
@@ -0,0 +1,445 @@
+---
+title: "<form>"
+canary: true
+---
+
+<Canary>
+
+React's `form` Component is currently only available in React's canary and experimental channels but you can use the [standard HTML `form` tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form) in your React's stable channel. Learn more about [React's release channels here](/community/versioning-policy#all-release-channels).
+
+</Canary>
+
+
+<Intro>
+
+The [built-in browser `<form>` component](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form) lets you create interactive controls for submitting information.
+
+```js
+<form action={search}>
+    <input name="query" />
+    <button type="submit">Search</button>
+</form>
+```
+
+</Intro>
+
+<InlineToc />
+
+---
+
+## Reference {/*reference*/}
+
+### `<form>` {/*option*/}
+
+The [built-in browser `<form>` component](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form) lets you create interactive controls for submitting information.
+
+```js
+<form action={search}>
+    <input name="query" />
+    <button type="submit">Search</button>
+</form>
+```
+
+[See more examples below.](#usage)
+
+#### 🚧 Props {/*props*/}
+
+`<form>` supports all [common element props.](/reference/react-dom/components/common#props)
+
+Additionally, `<form>` supports these props:
+
+* [`action`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/option#disabled): A boolean. Specifies what to do when a form is submitted. The form data can be handled with client or server side code.
+* 🚧
+
+
+🚧 Forms can contain one or more of the following tags:
+* [`<input>`]
+* [`<textarea>`]
+* [`<button>`]
+    * [`formAction`] prop that overrides `<form>`'s `action` prop
+* [`<select>`]
+* [`<option>`]
+* [`<optgroup>`]
+* [`<fieldset>`]
+* [`<label>`]
+* [`<output>`]
+* 🚧
+
+#### 🚧 Caveats {/*caveats*/}
+
+* 🚧
+
+---
+
+## Usage {/*usage*/}
+
+### 🚧 Submitting forms to the server {/*progressively-enhanching-forms*/}
+
+Render a `<form>` with a input and submit button. Pass a function marked with `'use server';` to the `action` prop of form to run the function when the form is submitted.
```

Comment:
Hmm, this is hard to explain and probably is dependent on the use server / use client docs. For example, if you have a file with "use client" and within that file you have a function with "use server", that won't work, right?

Reply:
Correct.

I started writing the "use server" docs and wanted to have a better idea of how forms were going to look like before I started on "use server" but they are going to be very interconnected in places like this. This explination was copied from the "Handling form submissions on the client" (which I wrote before this, and is itself a bad placeholder) and it needs to be completely rewritten.

I put this usage example first because I think that's what people are going to be most interested in: getting their form data to a server to do something with, but this example is just a placeholder because we don't have nodebox working. I think I figured out how to get all the other examples for `form` working on the client without nodebox but this example definitely needs nodebox to be a "proper" sample.

Reply:
Added a note on the relationship between server components and progressive enhancement